### ISSUE TITLE
`import/order` fixer breaks Svelte+TypeScript imports

### ISSUE THREAD

**Issue by ivanjonas** [import-js/eslint-plugin-import#2407]
This source:

```svelte
<script lang="typescript">
    import { AlertTypes, icons } from "./alert-types";
    import Icon from "ui/library/components/icon/icon.svelte";

    export let type = AlertTypes.Info;
</script>
```

produces one warning on each of the import lines. The first line produces "There should be at least one empty line between import groups - eslint (import/order)" and the second line produces "`ui/library/components/icon/icon.svelte` import should occur before import of `./alert-types` - eslint (import/order)".

Fixing the first warning works just fine. A newline is inserted between the imports:

```svelte
<script lang="typescript">
    import { AlertTypes, icons } from "./alert-types";

    import Icon from "ui/library/components/icon/icon.svelte";

    export let type = AlertTypes.Info;
</script>
```

Unfortunately, fixing the remaining warning with ESLint results in this broken output:

```svelte
<script lang="typescript">
    import { AlIcon from "ui/library/components/icon/icon.svelte";
import { AlertTypes, icons } from "./alert-types";


    export let type = AlertTypes.Info;
</script>
```

Note the double whitespace below the two imports. 

Any idea what's going on? My goal is to get the autofix to correctly swap those two import lines and not break the source code.

**Comment by ivanjonas** [comment]
Additional info: the same issue does not happen with a plain TypeScript file. 

```ts
import { icons } from "./alert-types";
import Icon from "ui/library/components/icon/icon.svelte";

const bob = Icon;
const tim = icons;

export { bob, tim };
```

The two import lines have the same warnings as in the original post above. I can fix them in any order and it does produce the expected result without breaking the source code. (The exports are just there to prevent "unused variable" warnings). Results after fixing are as expected:

```ts
import Icon from "ui/library/components/icon/icon.svelte";

import { icons } from "./alert-types";

const bob = Icon;
const tim = icons;

export { bob, tim };
```

**Comment by ljharb** [comment]
ok, so it seems like the svelte "processor" might be running, i assume prior to the TS parser?

I think to fix this we'd have to be able to add test cases that use the svelte processor, so that we can reproduce the failure inside our test system - and then we could figure out why it's failing (presumably, AST node name mismatch, or, a bug in svelte).

**Comment by ivanjonas** [comment]
Sounds complicated. Is there anything I can do to help?

**Comment by ljharb** [comment]
lol yes, i think it will be complicated unfortunately - once this infra is set up ofc, then future svelte processor problems will be much easier to fix/debug.

Other than making a PR for that, or finding someone willing to do it, I'm not sure what can be done.

**Comment by ivanjonas** [comment]
Oh boy. I might just have to give up on getting this bug fixed. Thankfully, it's the sort of thing that, once fixed across the whole code base, won't cause too much trouble in the day-to-day. Another alternative we're going to consider is to pop that code out of the <style> tag and into a dedicated .ts file, where the eslint rule works flawlessly.

**Comment by ljharb** [comment]
I’d probably suggest that anyways :-) that way it can be tested.

I’m happy to leave this open, though, because the bug should be fixed.

**Comment by Tofandel** [comment]
I'm having the same issue with the vue ts processor

```js
    'import/order': ['warn', {
      alphabetize: {
        order: 'asc',
        caseInsensitive: true,
      },
      'newlines-between': 'always',
    }],
 ```
 
 ```vue
 <script setup lang="ts">
import { useForm } from '@inertiajs/inertia-vue3';
import { useI18n } from 'vue-i18n';

import ValidationError from '@/Components/ValidationError.vue';

// ....
</script>
```

![image](https://user-images.githubusercontent.com/6115458/165301607-5cd56ae8-c07a-4141-a447-97f8bea2d580.png)

**Comment by ljharb** [comment]
@Tofandel are you sure it's not putting the third import in the same group as the second import?

**Comment by Tofandel** [comment]
@ljharb Yep I'm sure
![image](https://user-images.githubusercontent.com/6115458/165368838-44b23e8d-bebf-4e64-aa59-e2fb4740be6a.png)

It will also auto fix it, by adding a space in between, so I just removed `'newlines-between': 'always'`, don't know about the inner workings of the ts processor, is it maybe resolving this vue import as multiple imports? (One for template, one for script and one for style and eslint is running on this processed file?)

**Comment by ljharb** [comment]
Yes, that's possible. I'm not sure how the vue eslint parser handles that.

https://github.com/import-js/eslint-plugin-import/issues/2407#issuecomment-1075578624
> Other than making a PR for that, or finding someone willing to do it, I'm not sure what can be done.

**Comment by Airkro** [comment]
`import/order` fixer also beaks other processors. like <https://github.com/nice-move/eslint-plugin-html#not-compatible-with-importfirst-rule>

I believe it is the same problem. [reproduce sandbox](https://codesandbox.io/s/eslint-import-first-bug-mwdo7w?file=/index.html).

**Comment by callmeaponte** [comment]
FYI here's another instance where Svelte and this plugin don't play nicely together:

<br>

**Before**
```
<script context="module" lang="ts">
  import foo from "$lib/foo";
</script>
  
<script lang="ts">
  import bar from "@bar";
</script>
```

<br>

**After**
```
import bar from "@bar";
    import foo from "$lib/foo";
  
  

</script>
```

**Comment by ljharb** [comment]
@callmeaponte just like with vue, you'd need a svelte-specific eslint parser to be able to handle that.

**Comment by JounQin** [comment]
Use https://github.com/ota-meshi/eslint-plugin-svelte instead of hacky https://github.com/sveltejs/eslint-plugin-svelte.
